temporalio/worker/replayer.py

```diff
@@ -0,0 +1,272 @@
+"""Replayer."""
+
+from __future__ import annotations
+
+import concurrent.futures
+import copy
+import json
+import logging
+import re
+from typing import Any, Dict, Iterable, Optional, Type, Union
+
+import google.protobuf.json_format
+from typing_extensions import TypedDict
+
+import temporalio.api.history.v1
+import temporalio.bridge.worker
+import temporalio.converter
+
+from .interceptor import Interceptor
+from .worker import load_default_build_id
+from .workflow import _WorkflowWorker
+from .workflow_instance import UnsandboxedWorkflowRunner, WorkflowRunner
+
+logger = logging.getLogger(__name__)
+
+
+class Replayer:
+    """Replayer to replay workflows from history."""
+
+    def __init__(
+        self,
+        *,
+        workflows: Iterable[Type],
+        workflow_task_executor: Optional[concurrent.futures.ThreadPoolExecutor] = None,
+        workflow_runner: WorkflowRunner = UnsandboxedWorkflowRunner(),
+        namespace: str = "ReplayNamespace",
+        data_converter: temporalio.converter.DataConverter = temporalio.converter.default(),
+        interceptors: Iterable[Interceptor] = [],
+        build_id: Optional[str] = None,
+        identity: Optional[str] = None,
+        debug_mode: bool = False,
+    ) -> None:
+        """Create a replayer to replay workflows from history.
+
+        See :py:meth:`temporalio.worker.Worker.__init__` for a description of
+        arguments.
```

Comment:
You'll want to mention that this worker needs to use the same converters, interceptors, and has the same workflow type as the "real" worker that produced the history.

Reply:
I figured it was implied but I can be explicit.

Reply:
Added sentence.